api/maven-api-core/src/main/java/org/apache/maven/api/Constants.java

```diff
@@ -510,6 +510,22 @@ public final class Constants {
     @Config(type = "java.lang.Integer", defaultValue = "100")
     public static final String MAVEN_BUILDER_MAX_PROBLEMS = "maven.builder.maxProblems";
 
+    /**
+     * Configuration property for version range resolution used metadata "nature".
+     * It may contain following string values:
+     * <ul>
+     *     <li>"auto" - decision done based on range being resolver: if any boundary is snapshot, use "release_or_snapshot", otherwise "release"</li>
+     *     <li>"release_or_snapshot" - the default</li>
+     *     <li>"release" - query only release repositories to discover versions</li>
+     *     <li>"snapshot" - query only snapshot repositories to discover versions</li>
+     * </ul>
+     * Default (when unset) is existing Maven behaviour: "release_or_snapshots".
+     * @since 4.0.0
```

Comment:
4.1.0 ?

Reply:
@gnodet this needs backport to 4.0.0, but do I need to backport it manually or we have some automation?

Reply:
and is 4.0.0 as if you check https://github.com/apache/maven/pull/2574 the property is different: 3.x uses 3.x "way of naming", and 4.x already had `maven.versionResolver...` so this introduced `maven.versionRangeResolver...` prop

Reply:
> @gnodet this needs backport to 4.0.0, but do I need to backport it manually or we have some automation?

We don't.  I haven't been able to setup something that would work without a global GH access token as a secret.  I may have missed something though.
Anyway, I've been using [`backport`](https://github.com/sorenlouv/backport/tree/main) so far.